src/components/data-table.jsx

```diff
@@ -155,9 +152,10 @@ const DataTable = ({
   getIdKey,
   selectable,
   onHeaderClick,
+  compact,
```

Comment:
It would be good to have a `...propsForTable` to be able to pass any other prop to the underlying `<table>` tag.
It would need to merge a possible className prop like so:
```jsx
<table className={`data-table${compact ? ' data-table--compact' : ''}${propsForTable.className ? ` ${propsForTable.className}` : ''}`}>
```